### ISSUE TITLE
Error: Unable to resolve module `react-native/Libraries/promiseRejectionTrackingOptions`

### ISSUE THREAD

**Issue by FrankyLee-dev** [getsentry/sentry-react-native#1922]
react native：0.61.4
sentry：3.2.4

error: bundling failed: Error: Unable to resolve module `react-native/Libraries/promiseRejectionTrackingOptions` from `node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js`: react-native/Libraries/promiseRejectionTrackingOptions could not be found within the project.

**Comment by marandaneto** [comment]
@FrankyLee-dev thanks for reporting.

have you checked this? https://docs.sentry.io/platforms/react-native/troubleshooting/#unhandled-promise-rejections

**Comment by marandaneto** [comment]
@jennmueng could you have a look at this?
I could reproduce the build error by using `react-native init Name --version 0.61.4` and installing the SDK.

I've added:

```
  "resolutions": {
    "promise": "^7.1.1"
  }
```

to my App's package.json but the build error still persists.

**Comment by luisfuertes** [comment]
+1. App build but JS crash before init. 

```
error: bundling failed: Error: Unable to resolve module `react-native/Libraries/promiseRejectionTrackingOptions` from `node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js`: react-native/Libraries/promiseRejectionTrackingOptions could not be found within the project.
```

Platform Android
    "@sentry/react-native": "^3.2.4",
    "react": "16.9.0",
    "react-native": "0.61.5",

**Comment by marandaneto** [comment]
it works if using newer React Native versions, right?

**Comment by liucong1** [comment]
> +1. App build but JS crash before init.
> 
> ```
> error: bundling failed: Error: Unable to resolve module `react-native/Libraries/promiseRejectionTrackingOptions` from `node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js`: react-native/Libraries/promiseRejectionTrackingOptions could not be found within the project.
> ```
> 
> Platform Android "@sentry/react-native": "^3.2.4", "react": "16.9.0", "react-native": "0.61.5",
> 
> EDIT:
> 
> I try to add this with npm-force-resolutions but it didnt work
> 
> ```
>   "resolutions": {
>    "promise": "^7.1.1"
>  }
> ```

I have the same error, but @sentry/react-native works normally after being downgraded to 3.2.1

You can try
```
"@sentry/react-native": "3.2.1",
```

**Comment by luisfuertes** [comment]
If i downgrade to `"@sentry/react-native": "3.2.1",`

Error is:
```
error: bundling failed: Error: Unable to resolve module `promise/setimmediate/rejection-tracking` from `node_modules/@sentry/react-native/dist/js/integrations/reactnativeerrorhandlers.js`: promise/setimmediate/rejection-tracking could not be found within the project.
```

**Comment by marandaneto** [comment]
downgrade to https://github.com/getsentry/sentry-react-native/releases/tag/3.2.3 should be enough.
otherwise bumping the React Native SDK.
or keeping the version you had before, sorry for the inconvenience,  @jennmueng will look into it shortly.

**Comment by luisfuertes** [comment]
If downgrade to 3.2.3 have the same error as with the 3.2.1

**Comment by jennmueng** [comment]
This issue doesn't make sense and is especially concerning as the require that throws this error is inside a try catch block:

https://github.com/getsentry/sentry-react-native/blob/036c110811f182d87154c088d3792527e2bce558/src/js/integrations/reactnativeerrorhandlers.ts#L97-L115

and is supposed to be caught. In older RN versions this file isn't present and we would fallback to the default return in the catch block. Will look into this ASAP.

**Comment by nbpro** [comment]
"react-native": "0.63.4",
 "@sentry/react-native": "^3.2.4",
Getting this error after installing sentry for app.

**Comment by LFMAKER** [comment]
I tested it on version 0.63.4 of RN with the below versions of sentry SDK and got the same error.

Test with 
"@sentry/react-native": "3.2.4"
"@sentry/react-native": "3.2.3"
"@sentry/react-native": "3.2.1"

**Comment by johnsoncwb** [comment]
same error here, when i try to build or run on release mode (iOS) it close my app.

**Comment by sebasbeleno** [comment]
Same error here when i try to run dev environment on RN ^0.61.5

**Comment by jennmueng** [comment]
@LFMAKER @luisfuertes Make sure you are actually downgrading the SDK. Try a complete reinstall (delete node_modules etc) because the line of code was added in 3.2.4 it's technically impossible for any version prior to that to have this error.

Otherwise, I am investigating this issue as we speak.

**Comment by FrankyLee-dev** [comment]
> 

Yes, it is a runtime error.

I've added: 
```
  "resolutions": {
   "promise": "^7.1.1"
 }
```
but it didnt work.

@jennmueng The error has been fixed，i am using version 3.2.5.

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-react-native/releases/tag/3.2.5 fixes the issue, thank you all for the logs and sorry for the inconvenience.
gonna work on https://github.com/getsentry/sentry-react-native/issues/1924 soon to avoid such issues and detect them earlier.
